dandiapi/api/tasks.py

```diff
@@ -164,3 +167,57 @@ def validate_version_metadata(version_id: int) -> None:
 @shared_task
 def delete_doi_task(doi: str) -> None:
     delete_doi(doi)
+
+
+@shared_task
+def unembargo_dandiset(dandiset_id: int):
+    """Unembargo a dandiset by copying all embargoed asset blobs to the public bucket."""
+    dandiset: Dandiset = Dandiset.objects.get(id=dandiset_id)
+
+    # Only the draft version is needed, since embargoed dandisets can't be published
+    draft_version: Version = dandiset.draft_version
+    embargoed_assets: List[Asset] = list(
+        Asset.objects.filter(versions__in=[draft_version], embargoed_blob__isnull=False)
+    )
+
+    # Copy all embargoed assets to public bucket
+    for asset in embargoed_assets:
+        etag = asset.embargoed_blob.etag
+        matching_blob = AssetBlob.objects.filter(etag=etag).first() if etag != '' else None
```

Comment:
I'm not sure, I was being safe :man_shrugging: If you don't think it's a serious concern I can remove it.

Reply:
`BaseAssetBlob.etag = models.CharField(max_length=40, validators=[RegexValidator(f'^{ETAG_REGEX}$')])` if it were `""` it would be a SQL constraint violation, I don't think you need to check for it anywhere else.